codeworld-error-sanitizer/src/RegexShim.hs

```diff
@@ -38,23 +39,45 @@ replace regex replacement str = textFromJSString $
 
 #else
 
-import Data.Array (elems)
+import Data.Array (bounds, elems, (!))
+import Data.Char (isDigit)
 import Data.List
 import Data.Monoid
 import qualified Data.Text as T
+import Data.Text.Read (decimal)
 import Text.Regex.Base
 import Text.Regex.TDFA
 import Text.Regex.TDFA.Text
 
+getInserts :: Text -> [(Int, (Int, Int))]
+getInserts replacement = case replacement =~ ("[\\][0-9]" :: Text) :: [MatchArray] of
+    [] -> []
+    groups -> fmap processInsertion groups
+    where processInsertion group =
+              case decimal $ T.take len $ T.drop (start + 1) replacement of
```

Comment:
Did you mean `len - 1` here?  Right now, you're always looking one character beyond the matched expression, which seems confusing.  It sort of works because `decimal` succeeds when it matches only part of the string, but it will do strange things if the character after the group is still a number.

Reply:
Yes, you are right.